sharding/p2p/discovery.go

```diff
@@ -0,0 +1,71 @@
+package p2p
+
+import (
+	"context"
+	"time"
+
+	host "github.com/libp2p/go-libp2p-host"
+	peer "github.com/libp2p/go-libp2p-peer"
+	ps "github.com/libp2p/go-libp2p-peerstore"
+	mdns "github.com/libp2p/go-libp2p/p2p/discovery"
+	pb "github.com/prysmaticlabs/geth-sharding/sharding/p2p/proto/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+// Discovery interval for multicast DNS querying.
+var discoveryInterval = 1 * time.Minute
+
+// mDNSTag is the name of the mDNS service.
+var mDNSTag = mdns.ServiceTag
+
+// startDiscovery protocols. Currently, this supports discovery via multicast
+// DNS peer discovery.
+//
+// TODO: add other discovery protocols such as DHT, etc.
```

Comment:
Let's open up an issue for add other discovery protocols

Reply:
#287